awx/api/views/__init__.py

```diff
@@ -2957,6 +2957,18 @@ def is_valid_relation(self, parent, sub, created=False):
         if parent.id == sub.id:
             return {"Error": _("Cycle detected.")}
 
+        '''
+        Look for parent->child connection in all relationships except the relationship that is
+        attempting to be added; because it's ok to re-add the relationship
+        '''
+        relationships = ['success_nodes', 'failure_nodes', 'always_nodes']
+        relationships.remove(self.relationship)
+        qs = reduce(lambda x, y: (x | y),
+                    (Q(**{'{}__in'.format(rel): [sub.id]}) for rel in relationships))
```

Comment:
reduce is removed from the standard library in python3, something to keep in mind.